mat64/list_test.go

```diff
@@ -0,0 +1,684 @@
+								aCall = makeTransposeCopyOf(a).T()
+							}
+							if bTrans {
+								bCall = makeTransposeCopyOf(b).T()
+							}
+							method(receiver, aCall, bCall)
+						}
+
+						// Test the method for a zero-value of the receiver.
+						errStr := fmt.Sprintf("%T.%s(%T, %T), sizes: %#v, atrans %v, btrans %v", receiver, name, a, b, test, aTrans, bTrans)
+						zero := makeRandOf(receiver, 0, 0)
+						p, err := panics(func() { f(zero, a, b) })
+						if !legal && !p {
+							c.Errorf("Did not panic with illegal size: " + errStr)
+							continue
+						}
+						if legal && p {
+							c.Errorf("Panicked with legal size: " + errStr + err)
+							continue
+						}
+						if !equalApprox(zero, &want, 1e-14) {
+							c.Errorf("Answer mismatch with zero receiver:" + errStr)
+							continue
+						}
+						if !equal(a, aCopy) {
+							c.Errorf("First input argument changed in call: " + errStr)
+						}
+						if !equal(b, bCopy) {
+							c.Errorf("Second input argument changed in call: " + errStr)
+						}
+
+						if !legal {
+							continue
+						}
+
+						// Test the method with a non-zero-value of the receiver.
+						// The receiver has been overwritten in place so we can
+						// take its new size.
+						rr, rc := zero.Dims()
+						nonZero := makeRandOf(receiver, rr, rc)
+						panicked, _ := panics(func() { f(nonZero, a, b) })
+						if panicked {
+							c.Errorf("Panicked with non-zero receiver: " + errStr)
+						}
+						if !equalApprox(nonZero, &want, 1e-14) {
+							c.Errorf("Answer mismatch non-zero receiver: " + errStr)
+						}
+
+						// Test with an incorrectly sized matrix
+						switch receiver.(type) {
+						default:
+							panic("matrix type not coded for incorrect receiver size")
+						case *Dense:
+							wrongSize := makeRandOf(receiver, rr+1, rc)
+							panicked, _ = panics(func() { f(wrongSize, a, b) })
+							if !panicked {
+								c.Errorf("Did not panic with wrong number of rows: " + errStr)
+							}
+							wrongSize = makeRandOf(receiver, rr, rc+1)
+							panicked, _ = panics(func() { f(wrongSize, a, b) })
+							if !panicked {
+								c.Errorf("Did not panic with wrong number of columns: " + errStr)
+							}
+						case *TriDense, *SymDense:
+							// Only work for squares
+							wrongSize := makeRandOf(receiver, rr+1, rc+1)
+							panicked, _ = panics(func() { f(wrongSize, a, b) })
+							if !panicked {
+								c.Errorf("Did not panic with wrong size: " + errStr)
+							}
+						case *Vector:
+							// Must be a column
+							wrongSize := makeRandOf(receiver, rr+1, rc)
+							panicked, _ = panics(func() { f(wrongSize, a, b) })
+							if !panicked {
+								c.Errorf("Did not panic with wrong number of rows: " + errStr)
+							}
+						}
+
+						// TODO(btracey): Add test for Issue 147 if it is agreed that should be fixed.
+
+						// If the type and size of an input and ouput match, check
+						// that the function still works when one of them is the
+						// receiver
```

Comment:
Fixing this part of the code now. It doesn't do what it says it does for the transpose cases.